CONTRIBUTING.md

```diff
@@ -112,13 +112,17 @@ Just make sure to remove the dependencies again before finally making a pull req
 ### Rust Analyzer
 As of [#6869][6869], [`rust-analyzer`][ra_homepage] can understand that Clippy uses compiler-internals
 using `extern crate` when `package.metadata.rust-analyzer.rustc_private` is set to `true` in Clippy's `Cargo.toml.`
+
 You will require a `nightly` toolchain with the `rustc-dev` component installed.
+Notice that [rust-toolchain](./rust-toolchain) pins a specific nightly version, but for the
+configuration below to work you will need `rustc-dev` for the **regular** nightly.
```

Comment:
Hey!

I think I must have made a mistake during my first setup. Today I tried uninstalling `rustc-dev` from the nightly toolchain, did a `cargo clean`, recompiled everything and my rust-analyzer is working perfectly fine!

It may have something to do with the fact that for the past two days I experimented a little bit with some different ways of compiling clippy. So maybe at some point rust-analyzer was using the regular nightly instead of the pinned one. Another hypothesis that comes to mind is that I activated my rust-analyzer before creating the configuration mentioned in the documentation, so maybe in the first time I compiled it it wasn't using the correct settings and this caused problems.

But either way, thanks for pointing out that it wasn't needed. I can confirm that, now that I set everything up correctly, it works with just the pinned nightly's rustc-dev.